spec-examples/src/ntru_prime/ntru_prime.rs

```diff
@@ -0,0 +1,114 @@
+// Import hacspec and all needed definitions.
+use hacspec::prelude::*;
+
+
+/// Struct to decide NtruVersion
+pub struct NtruVersion{
+    pub p:i128,
+    pub q:i128,
+    pub w:i128,
+}
+#[macro_export]
+macro_rules! ntru_v {
+    ($t:expr)  => {
+        {
+        // TODO needs to be removed, just for demonstration
+        if $t == -1{
+            NtruVersion{p:191,q:397,w:68}
+        }
+        else if $t == 0{
+            NtruVersion{p:761,q:4591,w:286}
+        } else if $t == 1{
+            NtruVersion{p:653,q:4621,w:288}
+        } else {
+            NtruVersion{p:857,q:5167,w:322}
+        }
+
+        }
+    };
+
+}
+
+
+///This function creates a random polynom with w many -1 or 1 and with the highes degree of h_deg.
+fn create_rand_poly(w: i128, h_deg: usize) -> Vec<(usize, i128)> {
+    let mut counter = 0;
+    let mut pos;
+    let mut polynom: Vec<(usize, i128)> = Vec::new();
+    while w != counter {
+        pos = rand::thread_rng().gen_range(0, h_deg);
```

Comment:
Ok, so this needs library support. Can you open an issue for this? I don't think we want to allow this but only wrapped from the library.